Content.Server/Cargo/Systems/CargoSystem.Orders.cs

```diff
@@ -114,9 +115,12 @@ private void OnApproveOrderMessage(EntityUid uid, CargoOrderConsoleComponent com
                 return;
             }
 
-            // No order to approve?
-            if (!orderDatabase.Orders.TryGetValue(args.OrderIndex, out var order) ||
-                order.Approved) return;
+            // Find our order again. It might have been dispatched or approved already
+            var order = orderDatabase.Orders.Find(order => (args.OrderId == order.OrderId) && !order.Approved);
```

Comment:
How come the dictionary was removed if you're going to do the expensive find anyway?

Reply:
I haven't profiled this, but I wouldn't have classified this operation as "expensive" - it's O(N), for sure, but N is small; with the compiler inlining the lambda, your ICache, L2Cache, instruction pipelineing and branch prediction are all going to help out a bunch. I'd never expect to see this highlighted in a profile (could be wrong, obviously!) It's also a comparatively infrequent operation.

I got rid of the dictionary because:

1. I wanted the explicit ordering property of a List<T> (I would have used a Queue<T>, but sometimes this system wants to remove an item from the _middle_ of the order list) 
2. The old order-dispatch code was basically doing this same O(N) operation anyway in UpdateTelepad() and GetProjectedOrders()

If you're really worried about performance here, I can take a closer look and investigate. Certainly looks easy to make some improvements, but would really need to get some numbers.